### PR TITLE
feat: Settings page and report opt-out

### LINKED ISSUE #2099: Remove Bugsnag -- or allow to disable it at the very least.
## My Setup

- Operating System: any OS AFAIK
- App Version: 3.2.3 and older
- Installation type: any AFAIK

- [x] I have tested with the latest version
- [x] I can simulate the issue easily

## Description
Bugsnag should be removed from Rocket.Chat Electron or we should be able to completely disable it using a simple option.

### Current Behavior
Datas are being sent to third party WITHOUT user (or admin) approval.

### Expected Behavior
Please remove Bugsnag or allow to disable it.

I know that Rocket.Chat Electron is one of RC's part the least maintained but still, it feels really unacceptable to use Bugsnag like it is right now.

Related :
- #2005
- #1843

### PR DESCRIPTION
**Pull request by cazetto**
- Adds the settings screen
- Adds Report (Bugsnag) opt-out.

Closes #2099

### CODE REVIEW

File: src/ui/components/SettingsView/features/BugsnagOptIn.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    ({ isBugsnagEnabled }: RootState) => isBugsnagEnabled
+  );
+  const dispatch = useDispatch<Dispatch<RootAction>>();
+
+  const { t } = useTranslation();
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const isChecked = event.currentTarget.checked;
+    dispatch({ type: SETTINGS_SET_BUGSNAG_OPT_IN, payload: isChecked });
+  };
```
Comment (ggazzo) [inline src/ui/components/SettingsView/features/BugsnagOptIn.tsx:21]: useCallback

File: src/app/main/app.ts [hunk 2]
```diff
@@ -50,6 +53,22 @@ export const setupApp = (): void => {
...
+  if (isBugsnagEnabled) {
+    setupMainErrorHandling();
+  }
+
+  listen(SETTINGS_SET_BUGSNAG_OPT_IN, async () => {
+    app.relaunch();
+    app.quit();
+  });
+
+  // @TODO - Can I remove this line?
```
Comment (ggazzo) [inline src/app/main/app.ts:71]: try just disable/enable instead of relaunch

remove console.log

**Review by ggazzo — CHANGES_REQUESTED** [review]

File: src/ui/main/menuBar.ts [hunk 3]
```diff
@@ -471,6 +472,15 @@ const createWindowMenu = createSelector(
           dispatch({ type: SIDE_BAR_DOWNLOADS_BUTTON_CLICKED });
         },
       },
+      {
+        id: 'settings',
+        label: t('menus.settings'),
+        checked: currentView === 'settings',
+        accelerator: 'CommandOrControl+P',
```
[the remark this answers is not in the document]
Comment (cazetto) [reply src/ui/main/menuBar.ts:479]: What about cmd+i ?

File: src/app/main/persistence.ts [hunk 4]
```diff
@@ -5,7 +5,7 @@ import { PersistableValues, migrations } from '../PersistableValues';
 
 let electronStore: ElectronStore<PersistableValues>;
 
-const getElectronStore = (): ElectronStore<PersistableValues> => {
+export const getElectronStore = (): ElectronStore<PersistableValues> => {
```
Comment (cazetto) [inline src/app/main/persistence.ts:8]: stop exporting getElectronStore

File: src/preload.ts [hunk 5]
```diff
@@ -40,7 +39,8 @@ const start = async (): Promise<void> => {
 
   await whenReady();
 
-  setupRendererErrorHandling('webviewPreload');
+  // This removes the server error handling
+  // setupRendererErrorHandling('webviewPreload');
```
Comment (cazetto) [inline src/preload.ts:43]: is it common sense that this is going to be removed?

File: src/ui/components/SettingsView/features/BugsnagOptIn.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import { ToggleSwitch, Field } from '@rocket.chat/fuselage';
+import React, { ChangeEvent, Dispatch, FC, useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { RootAction } from '../../../../store/actions';
+import { RootState } from '../../../../store/rootReducer';
+import { SETTINGS_SET_BUGSNAG_OPT_IN } from '../../../actions';
+
+export const BugsnagOptIn: FC = () => {
```
Comment (cazetto) [inline src/ui/components/SettingsView/features/BugsnagOptIn.tsx:10]: Isn't it better to be called BugsnagOpt?

**Review by ggazzo — APPROVED** [review]

**Pull request merged**